packages/vitest/src/types/config.ts

```diff
@@ -843,6 +842,11 @@ export type ProjectConfig = Omit<
 > & {
   sequencer?: Omit<SequenceOptions, 'sequencer' | 'seed'>
   deps?: Omit<DepsOptions, 'moduleDirectories'>
+  poolOptions?: {
+    threads?: Pick<NonNullable<PoolOptions['threads']>, 'singleThread'>
+    vmThreads?: Pick<NonNullable<PoolOptions['vmThreads']>, 'singleThread'>
+    forks?: Pick<NonNullable<PoolOptions['forks']>, 'singleFork'>
+  }
```

Comment:
TODO: 

```suggestion
  poolOptions?: {
    threads?: Pick<NonNullable<PoolOptions['threads']>, 'singleThread' | 'isolate'>
    vmThreads?: Pick<NonNullable<PoolOptions['vmThreads']>, 'singleThread' | 'isolate'>
    forks?: Pick<NonNullable<PoolOptions['forks']>, 'singleFork' | 'isolate'>
  }
```

And make sure `poolOptions` from root `vitest.config.ts` is ignored. Pool is always initialized to use multiple workers and with isolation. Workspace projects can disable `isolation` in runner - not on pool level. 
Project level `singleThread`/`singleWorker` affects how many files are passed to Tinypool at a time.